src/lib/components/Pagination.svelte

```diff
@@ -0,0 +1,140 @@
+<script context="module" lang="ts">
+	const PAGE_SIZE = 50
+
+	export function paginate<T extends any[]>(pageParam: string | null, items: T) {
```

Comment:
Oh, I _really_ like this idea of having a `paginate` function that gets exported from the component. Very clever 😁 

I think that the `pageParam` should probably be an `int` so that the responsibility for parsing a potential query parameter is left to whomever calls this function rather than putting it in the function itself.